api/v1alpha1/amaltheasession_types.go

```diff
@@ -0,0 +1,265 @@
+	MountPath string `json:"mountPath,omitempty"`
+	// The secret containing the configuration or credentials needed for access to the data.
+	// The format of the configuration that is expected depends on the storage type.
+	// NOTE: define all values in a single key of the Kubernetes secret.
+	// rclone: any valid rclone configuration for a single remote, see the output of `rclone config providers` for validation and format.
+	SecretRef string `json:"secretRef,omitempty"`
+}
+
+type Culling struct {
+	// +kubebuilder:default:=0
+	// +kubebuilder:validation:Minimum:=0
+	// The maximum allowed age for a session, regardless of whether it
+	// is active or not. When the threshold is reached the session is hibernated.
+	// A value of zero indicates that Amalthea will not automatically hibernate
+	// the session based on its age.
+	MaxAgeSeconds int `json:"maxAgeSeconds,omitempty"`
+	// +kubebuilder:default:=0
+	// +kubebuilder:validation:Minimum:=0
+	// How long should a server be idle for before it is hibernated. A value of
+	// zero indicates that Amalthea will not automatically hibernate inactive sessions.
+	IdleSeconds int `json:"idleSeconds,omitempty"`
+	// +kubebuilder:default:=0
+	// +kubebuilder:validation:Minimum:=0
+	// How long can a server be in starting state before it gets hibernated. A
+	// value of zero indicates that the server will not be automatically hibernated
+	// by Amalthea because it took to long to start.
+	StartingSeconds int `json:"startingSeconds,omitempty"`
+	// +kubebuilder:default:=0
+	// +kubebuilder:validation:Minimum:=0
+	// How long can a server be in failed state before it gets hibernated. A
+	// value of zero indicates that the server will not be automatically
+	// hibernated by Amalthea if it is failing.
+	FailedSeconds int `json:"failedSeconds,omitempty"`
+	// +kubebuilder:default:=0
+	// +kubebuilder:validation:Minimum:=0
+	// Number of seconds where a server can be in hibernated state before
+	// it gets completely deleted. A value of zero indicates that hibernated servers
+	// will not be automatically be deleted by Amalthea after a period of time.
+	HibernatedSeconds int `json:"hibernatedSeconds,omitempty"`
+}
+
+// +kubebuilder:validation:Enum={token,oauth2proxy}
+type AuthenticationType string
+
+const Token AuthenticationType = "token"
+const Oidc AuthenticationType = "oauth2proxy"
+
+type Authentication struct {
+	Enabled bool               `json:"enabled,omitempty"`
+	Type    AuthenticationType `json:"type,omitempty"`
+	// Kubernetes secret that contains the authentication configuration
+	// For `token` generate a hard to guess string / password-like string.
+	// this value can be used as Authorization header or as a cookie with the name `amaltheaSessionToken` to
+	// access the session.
+	// For `oauth2proxy` please see https://oauth2-proxy.github.io/oauth2-proxy/configuration/overview#config-file.
+	SecretRef SecretRef `json:"secretRef,omitempty"`
```

Comment:
Looking at several objects that have SecretRef fields, it seems that all of them uses pointer for this, should we do the same ?